File: A7/ToyLanguage/Model/VarExp.cs
using System;
using ToyLanguage.Exceptions;

namespace ToyLanguage.Model
{
    public class VarExp : Exp
    {
        private String id;

        public VarExp(String id)
        {
            this.id = id;
        }
        public override int eval(IDict<String, int> dict)
        {
            if (!dict.isDefined(this.id)) throw new VariableNotFound();
            return dict.lookup(this.id);
        }
        public override string ToString()
        {
            return this.id; 
        }
    }
}

File: A8-ToyLanguageInterpretatorGUI/src/Model/VarExp.java
package Model;

import Exceptions.VariableNotFound;

public class VarExp extends Exp {
    private String id;

    public VarExp(String id) {
        this.id = id;
    }


    int eval(MyIDictionary<String, Integer> tbl, MyIRandIntKeyDict<Integer> heap) throws VariableNotFound {
        if(!tbl.isDefined(id)) throw new VariableNotFound();
        return tbl.lookup(id);
    }

    @Override
    public String toString() {
        return id;
    }


}